server/src/main/java/org/apache/iotdb/db/engine/trigger/executor/TriggerEngine.java

```diff
@@ -51,16 +57,24 @@ public static void fire(TriggerEvent event, InsertRowPlan insertRowPlan)
       if (mNode == null) {
         continue;
       }
-      TriggerExecutor executor = mNode.getTriggerExecutor();
-      if (executor == null) {
-        continue;
+      IMNode currentNode = mNode;
+      while (currentNode != null && !IoTDBConstant.PATH_ROOT.equals(currentNode.getName())) {
+        TriggerExecutor executor = currentNode.getTriggerExecutor();
+        currentNode = currentNode.getParent();
+        if (executor == null) {
+          continue;
+        }
+        executor.fireIfActivated(event, timestamp, values[i], mNode.getSchema().getType());
```

Comment:
Maybe we can avoid this mnode operation by adding interface like ```getInvolvedTrigger```. The mtree structure operation should be encapsulated in IMNode or schemaRegion.

Reply:
modified：
Class MNode add method 
public List<TriggerExecutor> getUpperTriggerExecutorList() {
    IMNode currentNode = this;
    List<TriggerExecutor> results = new ArrayList<>();
    while (currentNode != null && !IoTDBConstant.PATH_ROOT.equals(currentNode.getName())) {
      TriggerExecutor executor = currentNode.getTriggerExecutor();
      currentNode = currentNode.getParent();
      if (executor == null) {
        continue;
      }
      results.add(executor);
    }
    return results;
  }